rust/connlib/tunnel/src/tests/transition.rs

```diff
@@ -27,30 +27,27 @@ pub(crate) enum Transition {
     /// Client-side disable resource
     DisableResources(BTreeSet<ResourceId>),
     /// Send an ICMP packet to non-resource IP.
-    SendICMPPacketToNonResourceIp {
+    SendPacketToNonResourceIp {
```

Comment:
I am not sure if this unification is a good direction because I think we will have to unpack this again once we start testing additional properties of protocols on top of TCP or DNS. Once we have actual TCP connections here, I think a transition could be something like `SendHTTPRequest` or something like that which then takes a much bigger payload. Or we could have a `SendTcpPayload` that sends more bytes than what fits into a network packet.

Reply:
I consider it first keeping it separated but that leads to 9 transitions which seems a bit excessive, a compromise might be having ICMP/UDP in a single transition, which will probably be treated very similar and TCP in a separated one?

Reply:
Hmm, the non-resource IP ones made sense prior to the Internet Resource. I think we can just delete that one now?

Reply:
We can maybe also collapse the DNS and CIDR one. I don't remember why they are separate? 😅

Reply:
> Hmm, the non-resource IP ones made sense prior to the Internet Resource. I think we can just delete that one now?

But we don't always have an internet resource in the test, we still want to test behavior when no internet resource and sending outside our tests?

Reply:
> Hmm, the non-resource IP ones made sense prior to the Internet Resource. I think we can just delete that one now?

Yeah, that sounds good, we can have an enum to either domain name or IP, I think we already have something similar somewhere in the tests

Reply:
> What I meant was that the current transition already has to check dynamically, whether the packet gets routed. Having a separate transition made more sense when we could just ignore it. I guess checking for Internet Resource is still easier than actually checking for CIDRs too? Although with what the code looks like now, it may not be too difficult?

I'll take a stab at it, try to have a single `SendPayload` transition per-protocol